packages/types/src/jwt.ts

```diff
@@ -11,7 +11,7 @@ export interface JWTHeader {
   alg: string | Algorithm;
   typ?: string;
   cty?: string;
-  crit?: Array<string | Exclude<keyof JWTHeader, 'crit'>>;
+  crit?: Array<string | Exclude<JWTHeader, 'crit'>>;
```

Comment:
Does this mean previously this was broken ?

Reply:
@panteliselef based on the RFC: https://www.rfc-editor.org/rfc/rfc7515.html#section-4.1.11
The `crit` header parameter 
- should NOT be an empty array
- should not include `crit` (not sure about this but it's already defined)

The above change does not fix it, so i will update it to `crit: NonEmptyArray<Exclude<keyof JWTHeader, 'crit'>>;` to resolve this. I did some tests and it passes.

PS: `type NonEmptyArray<T> = [T, ...T[]];`